pkg/apis/scorecard/formatting.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"io"
+)
+
+type ScorecardFormatter interface {
+	MarshalText() (string, error)
+	MarshalJSONOutput(logReadWriter io.ReadWriter) ([]byte, error)
```

Comment:
I saw your commit about the `go vet` complaints. Another option might be to replace `MarshalJSONOutput` with an embedded interface for `json.Marshaller`, or maybe we can just rely on `json.Marshal` and not enforce that types explicitly define `MarshalJSON`. Thoughts?

Also, if we move the interface implementations as I mentioned here, would that help us avoid needing to pass the `io.ReadWriter` since we could set the overall run's log prior to formatting the output?

Reply:
in this latest commit, I opted to remove MarshalJSON() altogether, seemed to end in a simpler implementation to me.